Refactor bridge configuration

Lbqds opened the pull request:
Before this PR, if the contract address changes, we need to change the config of each module, the steps are:

1. Deploy Alephium/Ethereum contracts
2. Manually update contract addresses to `scripts/devnet-consts.json`
3. Update the configuration of each module, including `clients/js`, `bridge_ui`, `explorer`, `relayer/spy_relayer`, and `node`

After this PR, the steps are:

1. Deploy Alephium/Ethereum contracts
2. Run `npm run update_config` in the `configs` directory

Lbqds commented on Tiltfile:
```diff
@@ -273,17 +253,22 @@ if solana:
-  ignore = ["./artifacts", "./contracts", "./node_modules", "./dist", "./test"],
+  context = ".",
+  only = [
+    "./alephium/alephium-1.5.3.jar",
```
We don't need this, I added it because of my network problem, and forgot to remove it, thanks!

Lbqds commented on Tiltfile:
```diff
@@ -273,17 +253,22 @@ if solana:
+    "./alephium/contracts",
+    "./alephium/lib",
+    "./alephium/scripts",
+    "./alephium/auto-mine.sh",
```
I remembered that the consistent level for devnet is 10, I'll double-check it.

Lbqds replied:
The consistent level on devnet is 10: https://github.com/alephium/wormhole-fork/blob/dd41492ea7517b7e0d05dec9a55d5aea542fefdc/configs/alephium/devnet.json#L5

Lbqds commented on alephium/alephium.config.ts:
```diff
@@ -34,26 +49,23 @@ const configuration: Configuration<Settings> = {
-        ...settingsTemplate,
-        minimalConsistencyLevel: 10
-      }
+      settings: loadSettings('devnet')
```
Sounds great :+1, will fix.

Lbqds commented on devnet/alph-full-node.yaml:
```diff
@@ -33,97 +30,21 @@ spec:
+            - npx @alephium/cli devnet start;
+              npx @alephium/cli devnet stop;
+              echo 'alephium.network.internal-address="alph-full-node:19973"\nalephium.network.coordinator-address="alph-full-node:19973"' >> ~/.alephium-dev/user.conf;
+              ALEPHIUM_HOME=~/.alephium-dev/ java -jar ~/.alephium-dev/alephium*.jar &
```
The explorer backend doesn't work because of (this)[https://github.com/alephium/explorer-backend/pull/409], I'll remove this after this PR is merged.

polarker commented on clients/js/alph.ts:
```diff
@@ -21,24 +19,25 @@ import {
 ) {
-  const n = NETWORKS[network]['alephium']
-  if (!n.rpc) {
+  const n = CONFIGS[network]['alephium']
```
[Nit]maybe rename: `network -> networkType` and `n -> network`

But it seem single-character variable name is used sometimes in the codebase.

polarker commented on node/cmd/guardiand/node.go:
```diff
@@ -225,7 +225,7 @@ func init() {
 	alphRPC = NodeCmd.Flags().String("alphRPC", "", "Alephium RPC URL (required)")
 	alphApiKey = NodeCmd.Flags().String("alphApiKey", "", "Alphium RPC api key")
-	alphMinConfirmations = NodeCmd.Flags().Uint8("alphMinConfirmations", 10, "The min confirmations for alephium tx")
+	alphMinConfirmations = NodeCmd.Flags().Uint8("alphMinConfirmations", 1, "The min confirmations for alephium tx")
```
1. What is the default value for ETH?
2. Do you have a check somewhere for mainnet on the minimal number of confirmations?

Lbqds replied:
The default value for ETH is 1, the purpose of `minimalConfirmations` is that the guardian node will wait `max(minimalConfirmations, consistencyLevel)` blocks before signing the message, `consistencyLevel` is from the contract.

polarker reviewed (commented):
Great work ❤️ Just a few questions.

Lbqds commented on build-docker-images.sh:
```diff
@@ -18,40 +23,23 @@ else
 
 ## Build eth-node image
-pushd ethereum
-if [[ "$network" == 'devnet' ]] ; then
```
Can not find wormhole docker images in dockerhub, I'm not sure if they build docker images for mainnet and testnet.

Lbqds commented on ethereum/Dockerfile:
```diff
@@ -6,27 +6,19 @@ RUN apk add git python make build-base
-# npm
-RUN if [ -e /certs/cert.pem ]; then npm config set cafile /certs/cert.pem; fi
-# git
-RUN if [ -e /certs/cert.pem ]; then git config --global http.sslCAInfo /certs/cert.pem; fi
```
Because there is no CA files, I guess wormhole keep these CA files privately, and they add `**/cert.pem` to the `.gitignore` file, I'm not sure if we also need this.

polarker replied:
@h0ngcha0 Could you check if this is helpful/necessary for us?

h0ngcha0 replied:
I don't think we need this. This is probably needed if the ETH dev node tries to call other `https` endpoints that can not be verified with built-in certs. Since we don't find any issues with `devnet` and we will not use this image for `testnet` and `mainnet`, I think it is not necessary for us.

polarker commented:
@h0ngcha0 @Lbqds What are the differences between `function_server` and `cloud_function`? It seems they share quite some of the code.

Lbqds commented:
> @h0ngcha0 @Lbqds What are the differences between `function_server` and `cloud_function`? It seems they share quite some of the code.

The `cloud_function` is used to interact with BigTable, and the `function_server` is just an HTTP server, I guess we can remove the `function_server` now.

Lbqds commented:
I removed the `functions_server`, and I moved the `main.go` to the `server` directory because the package name of `main.go` must be `main`, so the compiler will compile it as an executable file instead of a library. cc @h0ngcha0
